### PR TITLE
[Android] Modify to use onboardingpayload library in CHIPTool

### PR DESCRIPTION
**Pull request by joonhaengHeo**
For using kotlin libary in Android Chiptool, I modified to use onboardingpayload library instead of setuppayload library.

For checking this library, some JNI difinition are wrong.
I also modified it.

### CODE REVIEW

**Comment by yufengwangca** [comment]
I just finished the phase I of onboardingpayload, and only update java-matter-controller to use onboardingpayload, and my original plan is to switch Android CHIPTool and Android TV to onboardingpayload after onboardingpayload is complete.

**Comment by yufengwangca** [comment]
I am doing the regression test since onboardingpayload is not finished yet, but it should works with Android CHIPTool

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt [hunk 1]
```diff
@@ -199,14 +196,14 @@ class BarcodeFragment : Fragment() {
...
+            ?.onCHIPDeviceInfoReceived(CHIPDeviceInfo.fromSetupPayload(payload))
     }
 
     private fun handleScannedQrCode(barcode: Barcode) {
         Handler(Looper.getMainLooper()).post {
-            lateinit var payload: SetupPayload
+            lateinit var payload: OnboardingPayload
             try {
-                payload = SetupPayloadParser().parseQrCode(barcode.displayValue)
+                payload = barcode.displayValue?.let { OnboardingPayloadParser().parseQrCode(it) } ?: return@post
```
Comment (yufengwangca) [inline examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt:206]: ?: Elvis operator is used to provide a default value or an alternative action when the expression before it is null. In this case, if displayValue is null, the code will immediately return from the current function or lambda, indicated by return@post.

Shall we assert or throw exception if displayValue is null?

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt [hunk 2]
```diff
@@ -36,11 +36,10 @@ import androidx.camera.lifecycle.ProcessCameraProvider
...
 import androidx.fragment.app.Fragment
-import chip.setuppayload.SetupPayload
-import chip.setuppayload.SetupPayloadParser
-import chip.setuppayload.SetupPayloadParser.SetupPayloadException
-import chip.setuppayload.SetupPayloadParser.InvalidEntryCodeFormatException
-import chip.setuppayload.SetupPayloadParser.UnrecognizedQrCodeException
+import chip.onboardingpayload.OnboardingPayload
+import chip.onboardingpayload.OnboardingPayloadParser
+import chip.onboardingpayload.OnboardingPayloadParser.OnboardingPayloadException
+import chip.onboardingpayload.OnboardingPayloadParser.UnrecognizedQrCodeException
```
Comment (yufengwangca) [inline examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt:42]: Those exceptions are also used by Generator part, I have moved them to chip.onboardingpayload from chip.onboardingpayload.OnboardingPayloadParser, I will update this part after this PR landed

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/QrCodeInfo.kt [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-import chip.setuppayload.OptionalQRCodeInfo.OptionalQRCodeInfoType
+import chip.onboardingpayload.OptionalQRCodeInfo.OptionalQRCodeInfoType
 import kotlinx.parcelize.Parcelize
 
 @Parcelize 
 data class QrCodeInfo(
     val tag: Int,
     val type: OptionalQRCodeInfoType,
-    val data: String,
+    val data: String?,
```
Comment (yufengwangca) [inline examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/QrCodeInfo.kt:11]: Do we have any case in which QrCodeInfo contains null data? I though this part is mandatory

Comment (joonhaengHeo) [reply examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/QrCodeInfo.kt:11]: https://github.com/project-chip/connectedhomeip/blob/a06bca80c1e214717c48d9fbe78a73fc2df42e31/src/controller/java/src/chip/onboardingpayload/OptionalQRCodeInfo.kt#L23

The part you implemented is nullable, so I made it the same.

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt (hunk 1 above)
Comment (joonhaengHeo) [reply examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt:206]: https://github.com/project-chip/connectedhomeip/blob/237784456a22ac51fe00c55179e27c8e58239974/examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt#L213

As you can see in this code, even if an exception occurs, it is terminated with return@post below. Therefore, I think it is better to return than to throw an exception at this location.

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt (hunk 2 above)
Comment (joonhaengHeo) [reply examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt:42]: I see.

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/QrCodeInfo.kt (hunk 3 above)
Comment (yufengwangca) [reply examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/QrCodeInfo.kt:11]: I think we should align java-matter-controller with Android CHIPTool QRCodeInfo object with a null data does not represent a valid state. I can update this in https://github.com/project-chip/connectedhomeip/pull/26963

**Review by yufengwangca — APPROVED** [review]

**Comment by andy31415** [comment]
@joonhaengHeo needs some conflict resolution

**Comment by joonhaengHeo** [comment]
I resolved some conflict.

**Pull request merged**